cmd/plugins/sgx-config/sgx-config.go

```diff
@@ -0,0 +1,184 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	"sigs.k8s.io/yaml"
+
+	"github.com/containerd/nri/pkg/api"
+	"github.com/containerd/nri/pkg/stub"
+)
+
+const (
+	// Key used for SGX annotations.
+	sgxEpcKey = "sgx-epc.nri.io"
+)
+
+var (
+	log     *logrus.Logger
+	verbose bool
+)
+
+// our injector plugin
+type plugin struct {
+	stub stub.Stub
+}
+
+// CreateContainer handles container creation requests.
+func (p *plugin) CreateContainer(_ context.Context, pod *api.PodSandbox, container *api.Container) (*api.ContainerAdjustment, []*api.ContainerUpdate, error) {
+	var (
+		ctrName string
+		sgxEpc  uint64
+		err     error
+	)
+
+	ctrName = containerName(pod, container)
+
+	if verbose {
+		dump("CreateContainer", "pod", pod, "container", container)
+	} else {
+		log.Infof("CreateContainer %s", ctrName)
+	}
+
+	adjust := &api.ContainerAdjustment{}
+
+	sgxEpc, err = parseSgxEpc(container.Name, pod.Annotations)
+	if err != nil {
+		log.Errorf("failed to parse SGX EPC annotation: %v", err)
+		return nil, nil, err
+	}
+
+	if sgxEpc > 0 {
+		adjust.AddLinuxUnified("misc.max", "sgx_epc "+strconv.FormatUint(sgxEpc, 10))
+
+		if verbose {
+			dump(ctrName, "ContainerAdjustment", adjust)
+		} else {
+			log.Infof("adjusted SGX EPC to %d", sgxEpc)
+		}
+	} else {
+		log.Infof("no SGX EPC annotations")
+	}
+
+	return adjust, nil, nil
+}
+
+func parseSgxEpc(ctr string, annotations map[string]string) (uint64, error) {
```

Comment:
As @askervin already mentioned in a [comment](https://github.com/containers/nri-plugins/pull/156#pullrequestreview-1681743401) the scope of this plugin will be maybe extended. Therefore the current function name `parseSgxEpc` could be misleading in the future. A more specific function name could be beneficial, e.g.:
```suggestion
// parseSgxEpcLimit returns defined EPC memory limit if defined.
func parseSgxEpcLimit(ctr string, annotations map[string]string) (uint64, error) {
```

Reply:
Updated this to `parseEpcLimit()` (also swapping the argument order the other way around which felt more natural (but probably only until I look at it the next time).